build.psm1

```diff
@@ -1192,8 +1192,36 @@ function Start-PSBootstrap {
             }
         }
 
-        $DotnetArguments = @{ Channel=$Channel; Version=$Version; NoSudo=$NoSudo }
-        Install-Dotnet @DotnetArguments
+        $dotNetExists = precheck 'dotnet' $null
+        $dotNetVersion = [string]::Empty
+        if($dotNetExists)
+        {
+            $dotNetVersion = (dotnet --version)
+        }
+
+        if(!$dotNetExists -or $dotNetVersion -ne '2.0.0-preview2-006502' -or $Force.IsPresent)
+        {
+            if($Force.IsPresent)
+            {
+                log "Installing dotnet due to -Force."
+
+            }
+            elseif(!$dotNetExistis)
+            {
+                log "dotnet not present.  Installing dotnet."
+            }
+            else
+            {
+                log "dotnet out of date ($dotNetVersion).  Updating dotnet."
+            }
+
+            $DotnetArguments = @{ Channel=$Channel; Version=$Version; NoSudo=$NoSudo }
+            Install-Dotnet @DotnetArguments
+        }
+        else
+        {
+            log "dotnet is already installed.  Skipping installation."
+        }
```

Comment:
The existing scripting style is to not put the opening curly brace in a new line, for example:
```
if ($IsWindows) {
...
So I think it's better to follow the same style.
```